src/libsync/propagateupload.h

```diff
@@ -355,25 +358,38 @@ class PropagateUploadFileNG : public PropagateUploadFileCommon
         quint64 size;
         QString originalName;
     };
-    QMap<int, ServerChunkInfo> _serverChunks;
+    QMap<quint64, ServerChunkInfo> _serverChunks;
+
+    // Linked list with expected PUT ranges.
+    struct ServerRangeInfo
+    {
+        quint64 start;
+        quint64 size;
+    };
+    QLinkedList<ServerRangeInfo> _serverRanges;
 
     /**
      * Return the URL of a chunk.
      * If chunk == -1, returns the URL of the parent folder containing the chunks
      */
-    QUrl chunkUrl(int chunk = -1);
+    QUrl chunkUrl(qint64 chunk = -1);
+    void updateRanges(quint64 start, quint64 size);
 
 public:
     PropagateUploadFileNG(OwncloudPropagator *propagator, const SyncFileItemPtr &item)
         : PropagateUploadFileCommon(propagator, item)
+        , _currentChunkSize(0)
```

Comment:
_currentChunkSize already has an initializer in the declaration

Reply:
Fixed.